java-generator/core/src/test/resources/io/fabric8/java/generator/approvals/ApprovalTest.generate_withValidCrd_shouldGeneratePojos.testAkkaMicroservicesCrd.approved.txt

```diff
@@ -99,6 +99,8 @@ public class AkkaMicroserviceSpec implements io.fabric8.kubernetes.api.model.Kub
         this.deploymentStrategy = deploymentStrategy;
     }
 
+    private static final org.test.v1.akkamicroservicespec.DeploymentStrategy _deploymentStrategy = io.fabric8.kubernetes.client.utils.Serialization.unmarshal("{\"type\":\"RollingUpdate\",\"rollingUpdate\":{\"maxSurge\":\"25%\",\"maxUnavailable\":\"25%\"}}", org.test.v1.akkamicroservicespec.DeploymentStrategy.class);
+
```

Comment:
Actually, now that I think about it, could this potentially lead to a problem because several instances have access to the same object and could invoke setters on it? Maybe I lead you into a rabbit hole with this optimization.

Feel free to back it out if you think it is problematic to share references to the same POJOs.

Reply:
Oh, wow! Thanks for this analysis @adriansuarez ! What you are saying makes sense if the user is registering classes to the ObjectMapper at runtime ...

I'm in doubt here, we can make the optimization optional behind a flag or we can decide that registering different classes from different threads without concurrency guards is already a bad enough practice that we don't want to support ... I'm open to hear feedback on when this is a common practice or happens to be a problem in real life projects.

Reply:
@andreaTP I think your original change is the best and safest due to the issue of the default objects being potentially mutable. We can't get make assumptions about the objects that are being deserialized here, so it is probably safest that each instance creates its own instances of the default fields as was the case before.

Reply:
@adriansuarez the more I think about this the more I'm convinced the current implementation is the best decision.

I fail to think about a real-world, legit, use-case where this encoding can cause trouble, but happy to look at an example that can change my mind :-)

Reply:
@adriansuarez I'm sorry to say that I fail to understand how this can happen.
The default value is hold in a "private static final" field, hence is immutable and not accessible, if you invoke "set" on one instance only that instance will be affected as the value will be set on the specific object instance, without affecting any other resource.

I will double check the code, but, if I'm missing a detail here I need to ask you to provide a compelling example so that I can understand were my reasoning is failing.

Reply:
ok, reverted to the original fixed encoding.

Reply:
Looks good to me. Thanks, @andreaTP !